src/lib/tabmanager.js

```diff
@@ -66,6 +84,18 @@ TabManager.prototype = {
       return;
     }
 
+    this.showGroup(chromeWindow, tabBrowser, groupID, tabIndex);
+  },
+
+  /**
+   * Show a given group.
+   *
+   * @param {ChromeWindow} chromeWindow
+   * @param {TabBrowser} tabBrowser
+   * @param {Number} groupID - the groupID
+   * @param {Number} tabIndex - the tab to activate
+   */
+  showGroup: function(chromeWindow, tabBrowser, groupID, tabIndex = 0) {
```

Comment:
I do not understand your intentions. Why did you split `selectGroup` and `showGroup`? Besides the naming is now really confusing, `selectGroup` is simply calling `showGroup` but doing a check if it is actually necessary to switch. Why do you need to skip those checks? You're only calling `showGroup` inside `moveTabToGroup` and I see no reason to not call `selectGroup` there.